File: Sword2offer/CH3-高质量代码/ES6/printNumRange.js
let range = function () {
  let nums = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
  function getRange (n) {
    if (n === 1) {
      return nums
    }
    let currentCount = n - 1
    let currentRange = []
    if (n >= 1) {
      let lastRange = getRange(currentCount)
      currentRange = nums.map((pre) => {
        return lastRange.map((tail) => {
          return pre + tail
        })
      })
      currentRange = currentRange.reduce((acc, value) => {
        return acc.concat(value)
      })
    }
    return currentRange
  }
  return function (n) {
    return getRange(n)
  }
}

let ranger = range()
console.log(ranger(3))

File: docs/Sword2offer/CH3-高质量代码/Python/printNumRange.py
class Ranger:
  def __init__(self, nums):
    self.nums = nums
  
  def __get_range(self, n):
    if n == 1:
      return self.nums
    current_range = []
    current_count = n - 1
    if current_count >= 1:
      last_range = self.__get_range(current_count)
      for pre in self.nums:
        current_row = []
        for tail in last_range:
          num = pre + tail
          current_row.append(num)
        current_range += current_row
    return current_range
  
  def get_range(self, n):
    return self.__get_range(n)

ranger = Ranger(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'])
print(ranger.get_range(3))
